### PR TITLE
fix(dropdownmenu): DropdownMenu now has mandatory props.onClickAway

### PR DESCRIPTION
**Pull request by tihuan**
## Summary

1. Update the scope `ClickAwayListener` listens to for better click away coverage
2. Install `czifui` locally via `yarn add -D ./`, so hopefully we don't need to manage `*.tgz`
3. Move Pagination `key` to the top component per linter`
4. Update snapshots accordingly
5. Bump `package.json` version due to new prop

Before:

If a user clicks on the same horizontal line as the InputDropdown component, the menu won't close

demo

After:

It now closes!

demo

## Checklist

- [ ] Default Story in Storybook
- [ ] LivePreview Story in Storybook
- [ ] Test Story in Storybook
- [ ] Tests written
- [ ] Variables from `defaultTheme.ts` used wherever possible
- [ ] If updating an existing component, depreciate flag has been used where necessary
- [ ] Chromatic build verified by @chanzuckerberg/sds-design

### CODE REVIEW

File: src/core/DropdownMenu/index.tsx [hunk 1]
```diff
@@ -119,75 +122,81 @@ const DropdownMenu = <
...
-            reason !== "reset" ||
-            (reason === "reset" && !keepSearchOnSelect)
-          ) {
-            setInputValue(value);
-          }
-          if (onInputChange) onInputChange(event, value, reason);
-        }}
-      />
-      {children}
+      <ClickAwayListener onClickAway={onClickAway}>
```
Comment (tihuan) [inline src/core/DropdownMenu/index.tsx:125]: Only change is to add `ClickAwayListener` and a wrapper `div`

File: src/core/DropdownMenu/index.stories.tsx [hunk 2]
```diff
@@ -214,132 +213,120 @@ const LivePreviewDemo = (): JSX.Element => {
   return (
     <div style={storyRow as React.CSSProperties}>
       <div style={{ alignSelf: "end", gridArea: "1/1/2/2" }}>
-        <ClickAwayListener onClickAway={handleClickAway1}>
```
Comment (tihuan) [inline src/core/DropdownMenu/index.stories.tsx:217]: remove ClickAwayListener and wrapper div - here and below

File: src/core/DropdownMenu/index.stories.tsx [hunk 3]
```diff
@@ -30,33 +30,32 @@ const Demo = (props: Args): JSX.Element => {
   const id = open ? `dropdown-menu` : undefined;
 
   return (
-    <ClickAwayListener onClickAway={handleClickAway}>
```
Comment (tihuan) [inline src/core/DropdownMenu/index.stories.tsx:33]: remove ClickAwayListener and add Fragment

File: src/core/InputDropdown/index.stories.tsx [hunk 4]
```diff
@@ -60,44 +59,41 @@ const Demo = (props: Args): JSX.Element => {
 
   return (
     <>
-      <ClickAwayListener onClickAway={handleClickAway}>
```
Comment (tihuan) [inline src/core/InputDropdown/index.stories.tsx:63]: remove ClickAwayListener and wrapper div

File: src/core/Pagination/index.tsx [hunk 5]
```diff
@@ -57,10 +57,9 @@ const Pagination = forwardRef<HTMLUListElement, PaginationProps>(
 
     return (
       <StyledPagination ref={ref} {...props}>
-        <StyledPaginationChevronList>
+        <StyledPaginationChevronList key="prevPage">
```
Comment (tihuan) [inline src/core/Pagination/index.tsx:60]: linter was flagging missing key on `StyledPaginationChevronList`

File: src/core/Dropdown/index.tsx [hunk 6]
```diff
@@ -111,71 +110,70 @@ const Dropdown = <Multiple extends boolean | undefined = false>({
   const [open, setOpen] = useState(false);
 
   return (
-    <ClickAwayListener onClickAway={handleClickAway}>
```
Comment (tihuan) [inline src/core/Dropdown/index.tsx:114]: Remove ClickAwayListener and add Fragment

File: src/core/ComplexFilter/index.tsx [hunk 7]
```diff
@@ -77,48 +76,41 @@ const ComplexFilter = <Multiple extends boolean | undefined = false>({
   // * such as sliders for ranges, inline multi selects, etc.
   return (
     <>
-      <ClickAwayListener onClickAway={handleClose}>
```
Comment (tihuan) [inline src/core/ComplexFilter/index.tsx:80]: Remove ClickAwayListener and wrapper div

File: package.json [hunk 8]
```diff
@@ -104,6 +104,7 @@
     "start": "start-storybook -p 6006",
     "build-storybook": "build-storybook -o docs-build",
     "test-storybook": "test-storybook",
+    "prelint": "yarn install-czifui-locally",
```
Comment (tihuan) [inline package.json:107]: seems like `lint` needs `czifui` installed as well, so I'm adding `prelint`. Does that sound right? Not sure why this wasn't needed before this PR 🤔

Comment (masoudmanson) [reply package.json:107]: Aah, thanks for adding this. 👍🏻

**Review by masoudmanson — APPROVED** [review]
Thank you @tihuan 
LGTM 🙇🏻‍♂️

**Pull request merged**